actors/builtin/storage_miner/storage_miner_actor.go

```diff
@@ -447,9 +424,7 @@ func (a *StorageMinerActor) _rtCheckTemporaryFaultEvents(rt Runtime, sectorNumbe
 
 	storageWeightDesc := a._rtGetStorageWeightDescForSector(rt, sectorNumber)
 
-	if checkSector.State == Active && rt.CurrEpoch() == checkSector.EffectiveFaultBeginEpoch() {
-		checkSector.State = TemporaryFault
-
+	if !st.IsSectorInTemporaryFault(sectorNumber) && rt.CurrEpoch() == checkSector.EffectiveFaultBeginEpoch() {
```

Comment:
I would feel much better if the `==` was `>=`. Ditto below.

Reply:
setting it to `>=` but noting that this check is scheduled on the CronActor and should be checking at the exact epoch.